test/alternator/test_batch.py

```diff
@@ -326,6 +326,18 @@ def test_batch_get_item_projection_expression_with_substitute_name(test_table):
         expected_items = [{k: item[k] for k in wanted if k in item} for item in items]
         assert multiset(got_items) == multiset(expected_items)
 
+# Same as above but we should return validation error when there is some unused name.
+def test_batch_get_item_projection_expression_with_unused_substitute_name(test_table):
+    items = [{'p': random_string(), 'c': random_string(), 'val2': random_string()} for i in range(10)]
+    with test_table.batch_writer() as batch:
+        for item in items:
+            batch.put_item(item)
+    keys = [{k: x[k] for k in ('p', 'c')} for x in items]
+    for wanted in [['p'], ['p', 'c'], ['val1'], ['p', 'val2']]:
+        with pytest.raises(ClientError, match='ValidationException'):
```

Comment:
As I noted in the previous patch, I don't think you need to repeat here all the complication of trying several combinations of attributes (which you copy-pasted from a different test) - it's enough to just test one which has "#unused" - we have no reason to suspect that it will work different for the case of "p" vs "p,c" vs. "p, val2" - none of those refers to "`#unused`".

If you want to add slightly more sophisticated checking for the unused attribute case, you can do a request which does use one name (e.g., `#ppp`) but doesn't use the other one `#unused`.

Reply:
simplified